src/metadata.rs

```diff
@@ -0,0 +1,33 @@
+use crate::storage_types::DataKey;
+use stellar_contract_sdk::{Env, Vec};
+
+pub fn read_decimal(e: &Env) -> u8 {
+    let key = DataKey::Decimals;
+    //TODO:should we return a default if entry doesn't exist
```

Comment:
You can use `e.has_contract_data`.

Reply:
Yeah but what value should be returned? 0? ERC20 using 18 by default because that's what ETH uses. Stellar assets use 7 so maybe this should be 7 as well?

Reply:
7 I guess. Is the goal of making it optional that you can save costs on deploying the contract?

Reply:
This should always be present the way the contract is currently written, so there is no need to handle it being absent. Please remove the TODO.

Reply:
TODO removed